shared/reducers/settings.tsx

```diff
@@ -78,7 +84,19 @@ function reducer(state: Types.State = initialState, action: SettingsGen.Actions)
     case SettingsGen.invitesClearError:
       return state.update('invites', invites => invites.merge({error: null}))
     case SettingsGen.loadedSettings:
-      return state.set('email', Constants.makeEmail({emails: action.payload.emails}))
+      return state
+        .set('email', Constants.makeEmail({emails: action.payload.emails}))
+        .set('phone', Constants.makePhone({phones: action.payload.phones}))
+    case EngineGen.keybase1NotifyEmailAddressEmailsChanged:
+      return state.setIn(
+        ['email', 'emails'],
+        I.Map(action.payload.params.list.map(row => [row.email, Constants.makeEmailRow(row)]))
```

Comment:
Need to check that `action.payload.params.list` exists first
```
list?: Array<Email> | null;
```
😓 ts `strictNullChecks` will fix this